lib/Sema/CSApply.cpp

```diff
@@ -6903,92 +6906,76 @@ static Expr *finishApplyDynamicCallable(ExprRewriter &rewriter,
     auto paramType = conformance->getTypeWitnessByName(
-        expectedParamType, cs.getASTContext().Id_ArrayLiteralElement)
-            ->getDesugaredType();
+        argumentType, cs.getASTContext().Id_ArrayLiteralElement);
 
     auto arrayElements =
         map<std::vector<Expr *>>(arg->getElements(), [&](Expr *origArgElt) {
-          return rewriter.coerceToType(origArgElt, paramType, loc);
+          return coerceToType(origArgElt, paramType, loc);
         });
-    auto *arrayExpr = ArrayExpr::create(
-        ctx, arg->getStartLoc(), arrayElements, {}, arg->getEndLoc());
-    cs.setType(arrayExpr, expectedParamType);
-    rewriter.finishArrayExpr(arrayExpr);
-    argument = arrayExpr;
+    argument = ArrayExpr::create(ctx, SourceLoc(), arrayElements,
+                                 {}, SourceLoc());
+    cs.setType(argument, argumentType);
+    finishArrayExpr(cast<ArrayExpr>(argument));
   } else {
-    auto dictLitProto = cs.TC.getProtocol(
-        fn->getLoc(), KnownProtocolKind::ExpressibleByDictionaryLiteral);
+    auto dictLitProto =
+        ctx.getProtocol(KnownProtocolKind::ExpressibleByDictionaryLiteral);
     auto conformance =
-        cs.TC.conformsToProtocol(expectedParamType, dictLitProto, cs.DC,
+        cs.TC.conformsToProtocol(argumentType, dictLitProto, cs.DC,
                                  ConformanceCheckFlags::InExpression);
-    auto keyAssocType =
-        conformance->getTypeWitnessByName(expectedParamType, ctx.Id_Key)
-            ->getDesugaredType();
-    auto valueAssocType =
-        conformance->getTypeWitnessByName(expectedParamType, ctx.Id_Value)
-            ->getDesugaredType();
-
+    auto keyType = conformance->getTypeWitnessByName(argumentType, ctx.Id_Key);
+    auto valueType = conformance->getTypeWitnessByName(argumentType,
+                                                       ctx.Id_Value);
+    SmallVector<Identifier, 4> names;
     SmallVector<Expr *, 4> dictElements;
     for (unsigned i = 0, n = arg->getNumElements(); i < n; i++) {
-      auto *labelExpr = new (ctx) StringLiteralExpr(
-          arg->getElementName(i).get(), arg->getElementNameLoc(i),
-          /*Implicit*/ true);
-      cs.setType(labelExpr, keyAssocType);
-
-      Expr *dictElt = TupleExpr::createImplicit(
-          ctx,
-          {rewriter.visitStringLiteralExpr(labelExpr),
-           rewriter.coerceToType(arg->getElement(i), valueAssocType, loc)},
-          {});
-      cs.setType(dictElt, TupleType::get({TupleTypeElt{keyAssocType},
-                                          TupleTypeElt{valueAssocType}},
-                                         ctx));
-      dictElements.push_back(dictElt);
-    }
-    auto *dictExpr = DictionaryExpr::create(
-        ctx, arg->getStartLoc(), dictElements, {}, arg->getEndLoc());
-    cs.setType(dictExpr, expectedParamType);
-    rewriter.finishDictionaryExpr(dictExpr);
-    argument = dictExpr;
+      Expr *labelExpr =
```

Comment:
Consider consistent 4-space indentation in this function body.

Reply:
Never mind - this seems to be exactly in sync with [`master`](https://github.com/apple/swift/blob/10a4e3ef41059608ffe9eded039350055064ce9f/lib/Sema/CSApply.cpp#L6814).